pkg/kubelet/rkt/pod_info.go

```diff
@@ -0,0 +1,225 @@
+	status.Name = container.Name
+	status.Image = container.Image
+
+	containerID, _ := parseContainerID(string(container.ID))
+	status.ImageID = containerID.imageID
+
+	switch p.state {
+	case Running:
+		// TODO(yifan): Get StartedAt.
+		status.State = api.ContainerState{
+			Running: &api.ContainerStateRunning{
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	case Embryo, Preparing, Prepared:
+		status.State = api.ContainerState{Waiting: &api.ContainerStateWaiting{}}
+	case AbortedPrepare, Deleting, Exited, Garbage:
+		exitCode, ok := p.exitCodes[status.ImageID]
+		if !ok {
+			glog.Warningf("rkt: Cannot get exit code for container %v", container)
+		}
+		exitCode = -1
+		status.State = api.ContainerState{
+			Termination: &api.ContainerStateTerminated{
+				ExitCode:  exitCode,
+				StartedAt: util.Unix(container.Created, 0),
+			},
+		}
+	default:
+		glog.Warningf("rkt: Unknown pod state: %q", p.state)
+	}
+	return status
+}
+
+// toPodStatus converts a podInfo type into an api.PodStatus type.
+func (p *podInfo) toPodStatus(pod *kubecontainer.Pod) api.PodStatus {
+	var status api.PodStatus
+	status.PodIP = p.ip
+	// For now just make every container's state the same as the pod.
+	for _, container := range pod.Containers {
+		status.ContainerStatuses = append(status.ContainerStatuses, p.getContainerStatus(container))
+	}
+	return status
+}
+
+// splitLineByTab breaks a line by tabs, and trims the leading and tailing spaces.
+// Not using strings.Split because there could be multiple tabs.
+func splitLineByTab(line string) []string {
+	var result []string
+	start := 0
+
+	line = strings.TrimSpace(line)
+	for i := 0; i < len(line); i++ {
+		if line[i] == '\t' {
+			result = append(result, line[start:i])
+			for line[i] == '\t' {
+				i++
+			}
+			start = i
+		}
+	}
+	result = append(result, line[start:])
+	return result
+}
+
+// getPodInfos returns a map of [pod-uuid]:*podInfo
+func (r *Runtime) getPodInfos() (map[string]*podInfo, error) {
+	output, err := r.runCommand("list", "--no-legend", "--full")
+	if err != nil {
+		return nil, err
+	}
+
+	if len(output) == 0 {
+		// No pods is running.
+		return nil, nil
+	}
+
+	// Example output of current 'rkt list --full' (version == 0.4.2):
+	// UUID                                 ACI     STATE      NETWORKS
+	// 2372bc17-47cb-43fb-8d78-20b31729feda	foo     running    default:ip4=172.16.28.3
+	//                                      bar
+	// 40e2813b-9d5d-4146-a817-0de92646da96 foo     exited
+	// 40e2813b-9d5d-4146-a817-0de92646da96 bar     exited
+	//
+	// With '--no-legend', the first line is eliminated.
+
+	result := make(map[string]*podInfo)
+	for _, line := range output {
+		tuples := splitLineByTab(line)
+		if len(tuples) < 3 { // At least it should have 3 entries.
+			continue
+		}
+		id := tuples[0]
+
+		status, err := r.runCommand("status", id)
+		if err != nil {
+			glog.Errorf("Cannot get status for pod (uuid=%q): %v", id, err)
+			continue
+		}
+		info := newPodInfo()
```

Comment:
WDYT: parsePodInfo(string) which returns the object, rather than creating and then parsing?